src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1160,22 +1160,6 @@ private Symbol UnwrapAlias(Symbol symbol, out AliasSymbol alias, BindingDiagnost
             else
             {
                 var boundTypeArguments = BindTypeArguments(typeArguments, diagnostics, basesBeingResolved);
-                if (unconstructedType.IsGenericType
-                    && options.IsAttributeTypeLookup())
-                {
-                    foreach (var typeArgument in boundTypeArguments)
-                    {
-                        var type = typeArgument.Type;
-                        if (type.IsUnboundGenericType() || type.ContainsTypeParameter())
```

Comment:
>type.IsUnboundGenericType() [](http://example.com/codeflow?start=28&length=27)

It looks like this check is dropped, not moved to ```CheckDisallowedAttributeDependentType````. It doesn't look like anything is obviously broken, but I would keep it anyway.

Reply:
The check is replaced by `type.Kind == SymbolKind.TypeParameter`. That's because previously, only the top-level types were traversed (e.g., `A<T>` and `B` in `A<T>.B`), but the `CheckDisallowedAttributeDependentType` method traverses the whole type (e.g., `T` also in `A<T>.B`). So after moving the checks there (https://github.com/dotnet/roslyn/pull/64006#discussion_r975027168), I changed them to fire on the type argument itself instead of on the `A<T>`.

https://github.com/dotnet/roslyn/blob/2fd652d9257dfcb245a738983bdf8364e769c0c2/src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs#L1369

Reply:
> The check is replaced by ```type.Kind == SymbolKind.TypeParameter```.

I am specifically referring to the ```type.IsUnboundGenericType()``` check. It is not the same as a type parameter check.

Reply:
Oh, right. I'm not sure how to hit that code path, but added nevertheless: d70a2ee8f113f9687651aa4d56bd37f0e6de58b5